File: cpp/alerts/scrape_reply.cpp
#include "./scrape_reply.h"

namespace ltrs {
    uint8_t scrape_reply_alert_get_version(lt::scrape_reply_alert *a) { return static_cast<uint8_t>(a->version); }
    int scrape_reply_alert_get_incomplete(lt::scrape_reply_alert *a) { return a->incomplete; }
    int scrape_reply_alert_get_complete(lt::scrape_reply_alert *a) { return a->complete; }
}

File: src/ffi/alerts/scrape_reply.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type scrape_reply_alert = crate::ffi::ffi::scrape_reply_alert;
    }
    unsafe extern "C++" {
        include!("cpp/alerts/scrape_reply.h");

        unsafe fn scrape_reply_alert_get_version(alert: *mut scrape_reply_alert) -> u8;
        unsafe fn scrape_reply_alert_get_incomplete(alert: *mut scrape_reply_alert) -> i32;
        unsafe fn scrape_reply_alert_get_complete(alert: *mut scrape_reply_alert) -> i32;
    }
}
